### DISCUSSION TITLE
Set schema name while first migration

### DISCUSSION THREAD

**Discussion by annndruha** [sqlalchemy/alembic#1272] [Usage Questions] [answered]
Set schema name while first migration. **Is there way to set what schema to use?**

**Databases / Backends / Drivers targeted**
PostgreSQL

**Example Use**
```
alembic init migration
```

**Additional context**
Failed because no select schema

**Comment by zzzeek** [comment, marked as answer]
step1: model is specific to a schema:

edit env.py here:

https://github.com/sqlalchemy/alembic/blob/main/alembic/templates/generic/env.py#L17-L21

set your model's `MetaData` there as described at https://alembic.sqlalchemy.org/en/latest/autogenerate.html#auto-generating-migrations 

then, in your model, set the schema name for each model, or the overall `MetaData`, as documented at https://docs.sqlalchemy.org/en/20/core/metadata.html#specifying-the-schema-name , as well as additional ORM-specific notes at https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#orm-declarative-table-schema-name 


step2: have autogenerate look in other schemas; see https://alembic.sqlalchemy.org/en/latest/autogenerate.html#omitting-schema-names-from-the-autogenerate-process for an example

set include_schemas=True: https://alembic.sqlalchemy.org/en/latest/api/runtime.html#alembic.runtime.environment.EnvironmentContext.configure.params.include_schemas

to determine *which* schemas to look at, if you have lots, set up an include_name hook: https://alembic.sqlalchemy.org/en/latest/api/runtime.html#alembic.runtime.environment.EnvironmentContext.configure.params.include_name

step3: for more involved multi-tenancy etc. see examples such as https://alembic.sqlalchemy.org/en/latest/cookbook.html#rudimental-schema-level-multi-tenancy-for-postgresql-databases

**Reply by annndruha** [reply]
Thank you very much!
